### DISCUSSION TITLE
CLI migrations com dotnet ef

### DISCUSSION THREAD

**Discussion by lucasmiko** [balta-io/forum#500] [Tira Dúvidas] [answered]
Estou no curso de fundamentos do aspnet 6 na aula Gerando o banco de dados da seção MVC, e não estou conseguindo criar os migrations por CLI, eu instalei de forma global o ef com o comando dotnet tool install --global dotnet-ef, mas quando uso o comando da aula para criar um migration tenho o seguinte warning:

![image](https://github.com/balta-io/forum/assets/63825231/b19d3aef-9048-45ee-8410-f21b35176667)

A versão do dotnet é:

![image](https://github.com/balta-io/forum/assets/63825231/01397a94-11b3-4a1a-a87c-f3b489f6f9b7)

Tentei usar o comando: dotnet tool install --global dotnet-ef --version 8.* mas recebo um alerta que o ef já está instalado.

Reiniciei a máquina mas o erro persiste. Como posso resolver?

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @lucasmiko, como vai?

Poderia executar o comando `dotnet ef --version` para vermos em qual versão ele está?

Em adicional, o comando `dotnet tool update --global dotnet-ef` atualiza as ferramentas do EF!

Elas devem estar na mesma versão do .NET!

Então por exemplo se você precisa usar o .NET 6, tem que instalar o EF com o comando:
`dotnet tool install --global dotnet-ef --framework net6.0`

**Reply by lucasmiko** [reply]
Bom dia @andrebaltieri , bem e você?

Eu desinstalei e instalei novamente, tentei usar o comando `dotnet tool install --global dotnet-ef --framework net8.0.101` então a versão instalada foi a 8.0.2:

![image](https://github.com/balta-io/forum/assets/63825231/325dafc0-782e-46a9-bb31-06a7dacccb96)

Após tentei verificar a versão mas o erro anterior foi exibido novamente:

![image](https://github.com/balta-io/forum/assets/63825231/d2bc58a9-eb5a-45cb-8879-dea64a73b192)

Então desinstalei o dotnet-ef de forma global e instalei novamente usando o `dotnet tool install --global dotnet-ef --framework net8.0`

Mas ainda não funcionou... Então acabei instalando localmente no projeto a versão 8.0.2, e então manualmente no arquivo dotnet-tools-json alterei para a versão 6.0:

![image](https://github.com/balta-io/forum/assets/63825231/5a9a8c7e-1fde-4ef8-aa3d-d9a27883809e)

Com isso no terminal executei um `dotnet tool restore`, a versão 6.0 foi reataurada e o comando da aula funcionou. 

![image](https://github.com/balta-io/forum/assets/63825231/5c381869-acc5-4b3b-8b3d-823e215ab8f4)


Obrigado! Deixei os detalhes aqui caso alguém mais precise.

**Reply by andrebaltieri** [reply]
Valeu demais @lucasmiko!! Certeza que vai ajudar outras pessoas 🚀
